pipelines/google_ads/generate_user_credentials.py

```diff
@@ -0,0 +1,195 @@
+"""This example will create an OAuth2 refresh token for the Google Ads API.
+
+This example works with both web and desktop app OAuth client ID types.
+
+https://console.cloud.google.com
+
+IMPORTANT: For web app clients types, you must add "http://127.0.0.1" to the
+"Authorized redirect URIs" list in your Google Cloud Console project before
+running this example. Desktop app client types do not require the local
+redirect to be explicitly configured in the console.
+
+Once complete, download the credentials and save the file path so it can be
+passed into this example.
+
+This example is a very simple implementation, for a more detailed example see:
+https://developers.google.com/identity/protocols/oauth2/web-server#python
+"""
+
+import argparse
+import hashlib
+import os
+import re
+import socket
+import sys
+from urllib.parse import unquote
+
+# If using Web flow, the redirect URL must match exactly what’s configured in GCP for
+# the OAuth client.  If using Desktop flow, the redirect must be a localhost URL and
+# is not explicitly set in GCP.
+from google_auth_oauthlib.flow import Flow
+
+_SCOPE = "https://www.googleapis.com/auth/adwords"
```

Comment:
why are we using the web flow not a desktop flow? we had nice script for google analytics. any reason?